singularity-eos/closure/mixed_cell_models.hpp

```diff
@@ -462,22 +462,87 @@ pte_closure_flag_with_line_impl(EOSIndexer &&eoss, const Real Volume, const Real
 // RealIndexer types may be different because some might be arrays and
 // some might be pointers.
 template <int nmat, typename T1, typename T2, typename T3>
-PORTABLE_INLINE_FUNCTION static void pte_residual(const Real utot, T1 &&vfrac, Real *u,
-                                                  T2 &&temp, T3 &&press, Real *residual) {
+PORTABLE_INLINE_FUNCTION static void pte_residual(const Real vfrac_tot, const Real utot,
+                                                  T1 &&vfrac, Real *u, T2 &&temp,
+                                                  T3 &&press, Real *residual) {
   Real vsum = 0.0;
   Real esum = 0.0;
   for (int m = 0; m < nmat; ++m) {
     vsum += vfrac[m];
     esum += u[m];
   }
-  residual[0] = 1.0 - vsum;
+  residual[0] = vfrac_tot - vsum;
   residual[1] = utot - esum;
   for (int m = 0; m < nmat - 1; ++m) {
     residual[2 + m] = press[m + 1] - press[m];
     residual[1 + nmat + m] = temp[m + 1] - temp[m];
   }
 }
 
+template <int nmat, typename EOSIndexer, typename RealIndexer, typename LambdaIndexer,
+          typename OFFSETTER>
+PORTABLE_INLINE_FUNCTION void
+try_ideal_pte(EOSIndexer &&eos, const Real vfrac_tot, const Real utot, const Real rho[nmat],
```

Comment:
I went with `try_ideal_pte` because the routine computes the PTE solution assuming ideal gases and then _might_ replace the initial guesses with that PTE solution depending on some criteria.  So, it tries to replace the initial guess with the ideal PTE solution and succeeds only sometimes, in general.  I agree, though, that I don't particularly like the name.  But it is slightly more descriptive that `get_ideal_pte`, at least as it's written right now.

Reply:
Yeah I saw that after reading further down... I'd like to split this into two functions... the acceptance criteria and the ideal gas PTE solution. But if you want to keep the function as a monolith the way it is currently, the name you have makes sense.

Reply:
A problem (not sure how serious this is of course) with that is that the scope of the trial state gets lifted into the caller.  Probably fine.  There's also a middle ground where I keep a `try_ideal_pte` which calls `get_ideal_pte` and then potentially overwrites the initial guess.  I'll try that.